### DISCUSSION TITLE
Split Window Difference RGB?

### DISCUSSION THREAD

**Discussion by bryanguarente** [blaylockbk/goes2go#53] [General]
Brian,
All of the following issues I am discussing are making some brash assumptions about the usage of Split Window Difference images.  You may have a different reason for making these images, and I am just missing that usage.  Proceed with caution and please take no offense if I am barking up the wrong tree.

I am working with some folks at the Alaska VAAC to make Split Window Difference images similar to the ones they use on the desk.  What I had assumed would work is to use your Split Window Difference method that is listed as an RGB in your code.  I think because you are assuming this is an RGB, the data gets normalized to between 0 and 1.  But in reality, the Split Window Difference is just that, a difference between two channels and shouldn't be normalized.  Typical values of the Split Window Difference images are between -6 and +5 for Volcanic Ash identification, -1 to +16 for low-level moisture assessments, and sometimes can be enhanced differently from -6 to +16 for low-level moisture assessments or dust identification.  By normalizing the data, the colorscales no longer work correctly in that "RGB".  

Then on top of the normalization issues, it looks like your calculation is also done backwards.  Your code is:
```python
data = ds["CMI_C15"].data - ds["CMI_C13"].data`
```
when it should be
```python
data = ds["CMI_C13"].data - ds["CMI_C15"].data
```
according to the RGB Quick Guide: http://cimss.ssec.wisc.edu/goes/OCLOFactSheetPDFs/ABIQuickGuide_SplitWindowDifference.pdf

I think this just shouldn't even be an RGB, but again, I may be barking up the wrong tree and you had another plan for this that I don't recognize.  

To plot this data, I just pulled the datasets with goes2go nearesttime and then did the following:
```python
first = cm.get_cmap('jet_r')
second = cm.get_cmap('binary_r')
newcolors = np.vstack((first(np.linspace(0, 1)), second(np.linspace(.5, 1))))
splitwindow_cmap = LinearSegmentedColormap.from_list("", newcolors)
splitwindow_cmap
im = ax.imshow(G['CMI_C13']-G['CMI_C15'], transform=G.rgb.crs, 
                               extent=[xdat.min(), xdat.max(), ydat.min(), ydat.max()], 
                               cmap=splitwindow_cmap, vmin=-6, vmax=5)
```
Hopefully all of this makes sense.  Feel free to ask questions if I didn't make this clear.  

Bryan

**Comment by blaylockbk** [comment]
Hi Bryan,
All your points are valid. You are right, the split difference is not an RGB and I should remove it from the RGB goes2go accessor to avoid any confusion. When I threw that in the code, I was just putting everything I could find into goes2go. I admit, I haven't used the split window for anything and haven't thought about it as thoroughly as you have.

I am starting to remember why I coded it that way...I wanted to provide the split window product in the RGB accessor since there were already many other products. I dumped the split window diff in the R, G, and B channels to make it display in greyscale, and the difference is backwards in my calculation to make the greyscale image not look inverted, and I normalized between 0-1 simply to fit the data into the RGB model. 6-years-ago me was just trying to make the split window diff easy to plot, but now that you brought this to my attention I know this isn't the correct way to do it. The method should be fixed and moved out of the RGB accessor.
